modules/repository-s3/src/yamlRestTest/java/org/elasticsearch/repositories/s3/RepositoryS3ClientYamlTestSuiteIT.java

```diff
@@ -8,20 +8,158 @@
 
 package org.elasticsearch.repositories.s3;
 
+import fixture.s3.junit.S3HttpFixtureRule;
+
 import com.carrotsearch.randomizedtesting.annotations.Name;
 import com.carrotsearch.randomizedtesting.annotations.ParametersFactory;
+import com.carrotsearch.randomizedtesting.annotations.ThreadLeakFilters;
 
+import org.elasticsearch.core.Booleans;
+import org.elasticsearch.core.SuppressForbidden;
+import org.elasticsearch.test.cluster.ElasticsearchCluster;
+import org.elasticsearch.test.cluster.local.LocalClusterSpecBuilder;
+import org.elasticsearch.test.cluster.util.resource.Resource;
+import org.elasticsearch.test.fixtures.minio.MinioFixtureTestContainer;
+import org.elasticsearch.test.fixtures.testcontainers.TestContainersThreadFilter;
 import org.elasticsearch.test.rest.yaml.ClientYamlTestCandidate;
 import org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase;
+import org.junit.ClassRule;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TestRule;
+
+import java.util.List;
+import java.util.Map;
+
+import static fixture.s3.junit.S3HttpFixtureRule.Type.S3Fixture;
+import static fixture.s3.junit.S3HttpFixtureRule.Type.S3FixtureEc2;
+import static fixture.s3.junit.S3HttpFixtureRule.Type.S3FixtureEcs;
+import static fixture.s3.junit.S3HttpFixtureRule.Type.S3FixtureSts;
+import static fixture.s3.junit.S3HttpFixtureRule.Type.S3FixtureWithToken;
 
+@ThreadLeakFilters(filters = { TestContainersThreadFilter.class })
 public class RepositoryS3ClientYamlTestSuiteIT extends ESClientYamlSuiteTestCase {
+    private static final boolean USE_FIXTURE = Booleans.parseBoolean(System.getProperty("tests.use.fixture", "true"));
+    private static final String TEST_TASK = System.getProperty("tests.task");
+    private static final S3HttpFixtureRule s3Environment = new S3HttpFixtureRule(USE_FIXTURE, requiredS3Fixtures());
+    private static final String s3TemporarySessionToken = "session_token";
 
-    public RepositoryS3ClientYamlTestSuiteIT(@Name("yaml") ClientYamlTestCandidate testCandidate) {
-        super(testCandidate);
+    @SuppressForbidden(reason = "Forbidden method invocation: java.lang.System#getProperties()")
+    private static List<S3HttpFixtureRule.Type> requiredS3Fixtures() {
+        String testTaskName = TEST_TASK.substring(TEST_TASK.lastIndexOf(':') + 1);
```

Comment:
Instead of having multiple test tasks why don't we consolidate this into a single task, and multiple test classes? I think we only modeled things this way because it was the only way to have different tests use different fixtures before. Let's ditch all these extra tasks and just create a test case which extends a base abtract test class for each of the individual scenarios.

Reply:
Initial I wanted to keep the surface of the change small. I go ahead and do that

Reply:
When running locally I see that those tests now still run in parallel. I thought we had that limited within a plugin...

Reply:
We allow for parallel execution of tests when using the new framework.